src/components/mols/CreditCardPayment/index.tsx

```diff
@@ -21,6 +23,18 @@ const CreditCardPayment: React.FC<ICreditCardPaymentProps> = ({
   const [loading, setLoading] = useState(false);
   const [formattedcreditCard, setFormattedcreditCard] = useState("");
   const Iugu = useIugu(process.env.NEXT_PUBLIC_IUGU_ID);
+  const { data: session, update } = useSession();
+
+  async function updateSession(newSubscriptionDate: string) {
+    await update({
+      ...session,
+      user: {
+        ...session?.user,
+        subscriptionEndDate: newSubscriptionDate,
+      },
+    });
+  }
```

Comment:
por que isso foi escrito assim e nao no padrão ES6:

```
const updateSession = async (newSubscriptionDate: string) => {
    ....
}
```


are we back to 2009 ou isso é obrigatorio ser assim por algum motivo que nao sei? 😆

Reply:
Foi só erro meu mesmo. É que tava escrito assim na primeira vez que usamos essa função e eu repeti o código.
inclusive talvez seja bom criar um util pra essa arrow funcion?
Atualizado para o ES6.